File: implementations/java/Summation.java
public class Summation {
    static final int SMALL  = 100;
    static final int MEDIUM = 10000;
    static final int LARGE  = 1000000;

    static long summation(long[] arr, int n) {
        long sum = 0;
        for (int i = 0; i < n; i++) sum += arr[i];
        return sum;
    }

    public static void main(String[] args) {
        int n;
        switch (args[0]) {
            case "small":  n = SMALL;  break;
            case "medium": n = MEDIUM; break;
            case "large":  n = LARGE;  break;
            default: System.err.println("Unknown size: " + args[0]); return;
        }
        long[] arr = new long[n];
        for (int i = 0; i < n; i++) arr[i] = i + 1;
        System.out.println(summation(arr, n));
    }
}

File: implementations/rust/summation.rs
use std::env;

const SMALL: usize  = 100;
const MEDIUM: usize = 10000;
const LARGE: usize  = 1000000;

fn summation(arr: &[i64], n: usize) -> i64 {
    let mut sum: i64 = 0;
    for i in 0..n {
        sum += arr[i];
    }
    sum
}

fn main() {
    let args: Vec<String> = env::args().collect();
    let n = match args[1].as_str() {
        "small"  => SMALL,
        "medium" => MEDIUM,
        "large"  => LARGE,
        _ => { eprintln!("Unknown size: {}", args[1]); return; }
    };
    let mut arr = vec![0i64; n];
    for i in 0..n {
        arr[i] = i as i64 + 1;
    }
    println!("{}", summation(&arr, n));
}
